pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -322,7 +322,7 @@ func (sb stageBuilder) isOutgoingOpEdgeAllowed(e *scgraph.OpEdge) bool {
 	if e.Type() != sb.opType {
 		return false
 	}
-	if !e.IsPhaseSatisfied(sb.bs.phase) {
+	if !e.IsPhaseSatisfied(sb.bs.phase) && !sb.bc.g.IsNoOp(e) {
```

Comment:
Nice. By the way, you probably should remove `hasOpEdgeWithOps` and the logic around it in this PR. Search for `// TODO(postamar): uphold the 2-version invariant using dep rules instead.`